### ISSUE TITLE
SC2086 should not complain about usage within set

### ISSUE THREAD

**Issue by smoser** [koalaman/shellcheck#1194]
It seems that SC2086 should not complain about usage with 'set'. In most cases if someone has unquoted strings in `set -- $VAR` then that was by design.  Perhaps there could be another warning code for use in `set --` specifically.

#### For bugs
- Rule Id: SC2086
- My shellcheck version: 0.4.6 and online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
foo="foo bar zee"
set -- $foo
echo "there were $# tokens in $foo"

```

#### Here's what shellcheck currently says:
    $ shellcheck myscript
     
    Line 3:
    set -- $foo
           ^-- SC2086: Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:
I'd like to not have to disable the check on that line and still get the generally-useful SC2086 globally.

**Comment by orbea** [comment]
You can avoid the warning with.
```
eval "set -- $foo"
```
However while this is safe in your example extra care should be taken to make sure you always control the variables when eval is used.

**Comment by koalaman** [comment]
I'm not sure this is a general issue with `set`. I'd argue for a warning even in this case because this doesn't seem intentional:

    there were 182 tokens in foo * zee

There are also other reasons why you'd want such a warning:

    var="hello world"
    set -- --msgbox $var 6 40
    dialog "$@"

If splitting/globbing is a generally an issue, but you've identified a specific case where it's not, then that's exactly what disable statements are for.
